src/LightSaml/Model/Protocol/Response.php

```diff
@@ -129,6 +130,10 @@ public function addEncryptedAssertion(EncryptedElement $encryptedAssertion)
 
     public function serialize(\DOMNode $parent, SerializationContext $context)
     {
+        if ($this->getID() === null) {
+            $this->setID(Helper::generateID());
+        }
```

Comment:
Why do you need to have an id in all cases ?

Reply:
The ID attribute is required for Response and Assertion, Right?